packages/components/src/confirm/hook.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import { useContextSystem, PolymorphicComponentProps } from '../ui/context';
+
+/**
+ * Internal dependencies
+ */
+import * as styles from './styles';
+import { useCx } from '../utils/hooks/use-cx';
+import type { OwnProps } from './types';
+
+export function useConfirm(
+	props: PolymorphicComponentProps< OwnProps, 'div' >
+) {
+	const { className, ...otherProps } = useContextSystem( props, 'Confirm' );
+
+	const cx = useCx();
+
+	const classes = cx( className );
+	const wrapperClassName = cx( styles.wrapper );
```

Comment:
There are a few improvements that can be applied here:
- Assuming that the `className` prop should be applied to the wrapper component, we should then apply `styles.wrapper` and `classname` in the same `cx()` call
- We should wrap the `cx()` inside a `useMemo` to reduce style recomputation from Emotion
- Following up on another inline comment (about avoiding inline styles in favour of applying styles with Emotion), this is where we could conditionally apply wrapper visibility styles


```
const wrapperClassName = useMemo(
	() =>
		cx(
			styles.wrapper,
			show && styles.wrapperHidden,
			className
		),
	[ className, show ]
);
```

Reply:
Ended up not needing to do that after I changed the implementation to use `Modal`.